### DISCUSSION TITLE
Group by alert and message formatting

### DISCUSSION THREAD

**Discussion by patrickklaeren** [datalust/seq-tickets#1649] [Help] [answered]
Given the following query in an alert:
```sql
select count(*) as count
from stream
where application is not null
group by time(1m), application
having count > 5
limit 100
```

How do I access the time/application/count from the grouped query to format the message in my alert. I have tried the following:

```
{results.count} or {results[0].count}
```

Both do not work.

**Comment by liammclennan** [comment, marked as answer]
Hi @patrickklaeren ,

The values available when processing alerts are described here: https://docs.datalust.co/docs/alert-properties

`Source.RangeStart` and `Source.RangeEnd` give the time. 

There is no 'application' value, since each record in the group may have a different `application` property (it is not an aggregate). If you choose to 'Include contributing events' in your alert definition then you can read the application properties of the included events. 

The count produced by the query is available in `Source.Results`.

**Reply by patrickklaeren** [reply]
Apologies if I am being a little slow here, but this does not work for me:

![image](https://user-images.githubusercontent.com/1341180/188096851-4412e052-8e3e-4877-a928-2c545d068a3b.png)

![image](https://user-images.githubusercontent.com/1341180/188096873-c1823d47-770f-4e00-a201-18de1fd720e5.png)

Is there something I'm missing here? Variables are being substituted as you can see, I have added a notification property called `Foobar`, so the templating itself works.

**Reply by patrickklaeren** [reply]
Ok thank you - update after you guided me to the alert schema/documentation.

I was barking up the wrong tree by changing the message of the alert, rather than just doing this directly in the app body. Have now got it working as intended. Thank you very much.

![image](https://user-images.githubusercontent.com/1341180/188131339-f1b94d86-fa8f-490e-aa21-9ff56320897d.png)

**Reply by liammclennan** [reply]
@patrickklaeren sorry for the slow reply. We are GMT +10 so it is Saturday here. I'm glad you found a solution, and thank you for posting an update.
